apps/coscrad-frontend/src/components/resources/spatial-features/leaflet/coscrad-leaflet-map.tsx

```diff
@@ -0,0 +1,41 @@
+import { LatLngExpression } from 'leaflet';
+import { MapContainer, TileLayer } from 'react-leaflet';
+import { INITIAL_CENTRE, INITIAL_ZOOM } from '../constants';
+import { CoscradMapProps, ICoscradMap } from '../map';
+import { buildSpatialFeatureMarker } from './spatial-feature-marker';
+
+export const CoscradLeafletMap: ICoscradMap = ({
+    spatialFeatures,
+    initialCentre,
+    initialZoom,
+    /**
+     * We inject the detail presenter to decouple the map from the presentation
+     * of the spatial feature in the popup. In the future, we could inject
+     * either the full-view or thumbnail view based on a config property.
+     */
+    DetailPresenter: spatialFeatureDetailPresenter,
+}: CoscradMapProps) => {
+    /**
+     * Not sure where to get this from, can it can be derived from the spatial feature coordinates to be displayed?
+     */
+    const initialMapCentreCoordinates: LatLngExpression = initialCentre || INITIAL_CENTRE;
+
+    const SpatialFeatureMarker = buildSpatialFeatureMarker(spatialFeatureDetailPresenter);
+
+    return (
+        <MapContainer
+            center={initialMapCentreCoordinates}
+            zoom={initialZoom || INITIAL_ZOOM}
+            scrollWheelZoom={false}
+        >
+            <TileLayer
+                attribution="&copy; ESRI and Contributors"
+                // Should this be part of the config?
```

Comment:
This is a great question, would be good to have it as a config.  This is also where in the future it would be great to be able to choose map media to be able to implement locally made maps.  There's also the GDAL library and MapTiler that could be used to parse a high res image uploaded into map tiles.